OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/SecurityTokenHelper.java

```diff
@@ -756,6 +756,126 @@ public String getUserId() throws IOException {
         return signature;
     }
 
+    /**
+     * Call INTERNAL AUTHENTICATE command and returns the MPI value
+     *
+     * @param hash the hash for signing
+     * @return a big integer representing the MPI for the given hash
+     */
+    public byte[] calculateAuthSignature(byte[] hash, int hashAlgo) throws IOException {
```

Comment:
to prevent code duplication: everything that is similar to what we have in ``calculateSignature`` should move into a different method, like the construction of the ``dsi``